src/wallet/wallet.cpp

```diff
@@ -1619,18 +1620,20 @@ int64_t CWallet::RescanFromTime(int64_t startTime, const WalletRescanReserver& r
  * from or to us. If fUpdate is true, found transactions that already
  * exist in the wallet will be updated.
  *
- * Returns null if scan was successful. Otherwise, if a complete rescan was not
- * possible (due to pruning or corruption), returns pointer to the most recent
- * block that could not be scanned.
+ * @param[in] pindexStop if not a nullptr, the scan will stop at this block-index
+ * @param[out] failed_block if FAILURE is returned, will be set to the most
+ *    recent block that could not be scanned.
```

Comment:
nit: could return `last_block`, which will either be "the most recent block that could not be scanned" (failure/abort) or "pindexStop/::chainActive.Tip()" (success)